### PR TITLE
Websocket RPC connection

### PR DESCRIPTION
**Pull request by kerzhner**
Fixes https://github.com/statechannels/go-nitro/issues/1097.

This PR introduces the websocket connection implementation that adheres to the same interface as the nats connection implementation.

### CODE REVIEW

File: rpc/transport/ws/client.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		err = json.Unmarshal(data, &unmarshaledNotifcation)
+		if err != nil {
+			panic(err)
+		}
+		wsc.logger.Trace().Msgf("Received message: %v", unmarshaledNotifcation)
+
+		// Is this a notification?
+		if unmarshaledNotifcation.Method != "" {
+			wsc.notificationChan <- data
+			// Or is this a reply?
```
Comment (geoknee) [inline rpc/transport/ws/client.go:76]:
```suggestion
			// Or is this a response?
```

File: rpc/transport/ws/client.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	if err != nil {
+		return nil, err
+	}
+	wsc.clientWebsocket = conn
+	go func() { wsc.readMessages(context.Background()) }()
+	return wsc, nil
+}
+
+func (wsc *clientWebSocketConnection) Request(id uint64, data []byte) ([]byte, error) {
+	// Any request payload type will do here since we are only interested in the id
```
Comment (geoknee) [inline rpc/transport/ws/client.go:35]: Not sure I follow this comment (or the others like it in this file)

File: rpc/transport/ws/server.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+func (wsc *serverWebSocketConnection) Close() {
+	err := wsc.httpServer.Shutdown(context.Background())
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (wsc *serverWebSocketConnection) Url() string {
+	return "ws://127.0.0.1:" + wsc.port
```
Comment (geoknee) [inline rpc/transport/ws/server.go:72]: Is hardcoding a local address a bit brittle? Should we at least make it a constant?

**Review by geoknee — APPROVED** [review]
🌋

**Comment by kerzhner** [comment]
Rebasing this work onto main is complex. Opening https://github.com/statechannels/go-nitro/pull/1132 instead.

File: rpc/transport/ws/client.go (hunk 1 above)
Comment (kerzhner) [reply rpc/transport/ws/client.go:76]: Changed in [a7aee17](https://github.com/statechannels/go-nitro/pull/1132/commits/a7aee176ccf6ad1d3f3600b1aefdb1b45c3d8db0)

File: rpc/transport/ws/server.go (hunk 3 above)
Comment (kerzhner) [reply rpc/transport/ws/server.go:72]: Good call. The address is now a constant in 28110881

File: rpc/transport/ws/client.go (hunk 2 above)
Comment (kerzhner) [reply rpc/transport/ws/client.go:35]: Agreed that these comments are confusing. We no longer have "any" request/response types. So these comments are removed.

**Comment by kerzhner** [comment]
@geoknee Comments from this PR have been addressed in the new PR. The new PR's diff is mostly the same as this PR diff. [62e14bd](https://github.com/statechannels/go-nitro/pull/1132/commits/62e14bd210b23e43544dc54edb64df3251a1c11c) was introduced in the new PR to account for changes in https://github.com/statechannels/go-nitro/pull/1126

**Pull request closed without merging**